src/combinators/mod.rs

```diff
@@ -1,164 +1,123 @@
 use crate::gen::GenError;
 use crate::lib::std::io::{self, SeekFrom, Write};
+use crate::lib::std::io::Seek as _;
 
 /// Holds the result of serializing functions
 ///
 /// The `Ok` case returns the `Write` used for writing, in the `Err` case an instance of
 /// `cookie_factory::GenError` is returned.
-pub type GenResult<I> = Result<I, GenError>;
+pub type GenResult<I> = Result<WriteContext<I>, GenError>;
 
 /// Trait for serializing functions
 ///
 /// Serializing functions take one input `I` that is the target of writing and return an instance
 /// of `cookie_factory::GenResult`.
 ///
 /// This trait is implemented for all `Fn(I) -> GenResult<I>`.
-pub trait SerializeFn<I>: Fn(I) -> GenResult<I> {}
+pub trait SerializeFn<I>: Fn(WriteContext<I>) -> GenResult<I> {}
 
-impl<I, F: Fn(I) -> GenResult<I>> SerializeFn<I> for F {}
+impl<I, F: Fn(WriteContext<I>) -> GenResult<I>> SerializeFn<I> for F {}
 
-/// Trait for `Write` types that allow skipping over the data
-pub trait Skip: Write {
-    fn skip(self, s: usize) -> Result<Self, GenError> where Self: Sized;
+pub struct WriteContext<W> {
+    write: W,
+    position: u64,
 }
```

Comment:
You could just make the two fields public and drop the accessors + only keep new/into_inner.

You even could drop the new and just impl From<W>

Reply:
Makes sense, good idea!